source/security/authentication.txt

```diff
@@ -0,0 +1,401 @@
+  it in a connection URI.
+- ``authSource``: The MongoDB database to authenticate against. By default, the
+  {+php-library+} authenticates against the ``admin`` database.
+- ``authMechanism``: Set to ``'SCRAM-SHA-1'``.
+
+You can set these options in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through parameters in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $options = [
+             'username' => '<username>',
+             'password' => '<password>',
+             'authSource' => '<authentication database>',
+             'authMechanism' => 'SCRAM-SHA-1',
+         ];
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            $options,
+         );
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<username>:<password>@<hostname>:<port>/?authSource=admin&authMechanism=SCRAM-SHA-1';
+         $client = new Client($uri);
+
+.. _php-mongodb-x509:
+
+MONGODB-X509
+------------
+
+If you enable TLS, during the TLS handshake, the {+php-library+} can present an X.509
+client certificate to MongoDB to prove its identity. The MONGODB-X509 authentication
+mechanism uses this certificate to authenticate the client.
+
+To authenticate with this mechanism, set the following connection options:
+
+- ``tls``: Set to ``true``.
+- ``tlsCertificateKeyFile``: The file path of the ``.pem`` file that contains your
+  client certificate and private key.
+- ``authMechanism``: Set to ``'MONGODB-X509'``.
+
+You can set these options in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through parameters in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $options = [
+             'tls' => true,
+             'tlsCertificateKeyFile' => '<file path>',
+             'authMechanism' => 'MONGODB-X509',
+         ];
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            $options,
+         );
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<hostname>:<port>/?tls=true&tlsCertificateKeyFile=<file path>&authMechanism=MONGODB-X509';
+         $client = new Client($uri);
+
+.. _php-mongodb-aws:
+
+MONGODB-AWS
+-----------
+
+.. important::
+
+   The MONGODB-AWS authentication mechanism requires {+mdb-server+} v4.4 or later.
+
+The MONGODB-AWS authentication mechanism uses AWS IAM (Amazon Web Services Identity and
+Access Management) or AWS Lambda credentials to authenticate your application. To use this
+method to authenticate, create a user with an associated Amazon Resource
+Name (ARN) on the ``$external`` database. Then, specify ``'MONGODB-AWS'`` as the value of
```

Comment:
I have only a basic understanding of the `MONGODB-AWS` mechanism, but I don't think this is correct. `$external` is also the default auth source used by `MONGODB-X509`, and it implies that the credentials are coming from an external source. There is no actual database named `$external`.

In fact, MongoDB doesn't allow databases to include a "$" character (see: [Naming Restrictions](https://www.mongodb.com/docs/manual/reference/limits/#naming-restrictions)).

I'll note that this text doesn't appear in the corresponding [PyMongo docs](https://www.mongodb.com/docs/languages/python/pymongo-driver/current/security/authentication/#mongodb-aws).

Reply:
This is based on the [C++ page](https://www.mongodb.com/docs/languages/cpp/cpp-driver/upcoming/security/authentication/#mongodb-aws) but my understanding of AWS auth and $external is also limited. I changed the wording to something more generic, but let me know your thoughts

Reply:
The C driver page inconsistently says "`$external` virtual database" and "`$external` database". I think including "virtual" makes sense.

I see you removed all mention of "$external" in this tutorial, which seems ideal.